pkg/operator/ceph/csi/operator_config.go

```diff
@@ -0,0 +1,162 @@
+	opcontroller "github.com/rook/rook/pkg/operator/ceph/controller"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "k8s.io/api/core/v1"
+	k8scsiv1 "k8s.io/api/storage/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const opConfigCRName = "ceph-csi-operator-config"
+
+func (r *ReconcileCSI) CreateOrUpdateOpConfig(csiClusterConfig *cephclient.ClusterInfo, cluster cephv1.CephCluster) error {
+	logger.Info("Creating ceph-CSI operator config CR")
+
+	opConfig := &csiopv1a1.OperatorConfig{}
+	opConfig.Name = opConfigCRName
+	opConfig.Namespace = r.opConfig.OperatorNamespace
+	opConfig.Spec = *r.CreateOpConfigSpec(csiClusterConfig, cluster, opConfig)
+
+	err := r.client.Create(r.opManagerContext, opConfig)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return errors.Wrapf(err, "failed to create CSI-operator operator config CR %s", opConfig.Name)
+		}
+		logger.Infof("CSI-operator operator config cr %s already exist, updating...", opConfig.Name)
+		err := r.client.Get(r.opManagerContext, types.NamespacedName{Name: opConfigCRName, Namespace: r.opConfig.OperatorNamespace}, opConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get CSI-operator operator config CR %s", opConfigCRName)
+		}
+
+		opConfig.Spec = *r.CreateOpConfigSpec(csiClusterConfig, cluster, opConfig)
+
+		err = r.client.Update(r.opManagerContext, opConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update CSI-operator operator config CR %s", opConfig.Name)
+		}
+
+		logger.Info("Successfully updated ceph-CSI operator config CR")
+	}
+
+	return nil
+}
+
+func (r *ReconcileCSI) CreateOpConfigSpec(csiClusterConfig *cephclient.ClusterInfo, cluster cephv1.CephCluster, opConfig *csiopv1a1.OperatorConfig) *csiopv1a1.OperatorConfigSpec {
+	maxLogSize, _ := opcontroller.GetLogRotateConfig(cluster.Spec)
+	cephfsClientType := csiopv1a1.KernelCephFsClient
+	if CSIParam.ForceCephFSKernelClient == "false" {
+		cephfsClientType = csiopv1a1.FuseCephFsClient
+	}
+
+	var periodicity csiopv1a1.PeriodicityType
+	switch cluster.Spec.LogCollector.Periodicity {
+	case "1h", "hourly":
+		periodicity = csiopv1a1.HourlyPeriod
+	case "weekly":
+		periodicity = csiopv1a1.WeeklyPeriod
+	case "monthly":
+		periodicity = csiopv1a1.MonthlyPeriod
+	default:
+		periodicity = csiopv1a1.DailyPeriod
+	}
+
+	opConfig.Spec = csiopv1a1.OperatorConfigSpec{
+		LogLevel: int(CSIParam.LogLevel),
+		DriverSpecDefaults: &csiopv1a1.DriverSpec{
+			Log: &csiopv1a1.LogSpec{
+				LogLevel: int(CSIParam.LogLevel),
+				Rotation: &csiopv1a1.LogRotationSpec{
+					MaxLogSize:  maxLogSize,
+					Periodicity: periodicity,
+				},
+			},
+			//  ImageSet: ,
+			ClusterName:    &cluster.Name,
+			EnableMetadata: &CSIParam.CSIEnableMetadata,
+			GRpcTimeout:    int(CSIParam.GRPCTimeout),
+			// SnapshotPolicy: ,
+			GenerateOMapInfo: &CSIParam.EnableOMAPGenerator,
+			FsGroupPolicy:    k8scsiv1.FileFSGroupPolicy,
+			// Encryption: ,
```

Comment:
if `EnableCSIEncryption` is set we need to set `rook-ceph-csi-kms-config` as the configmap name